Cargo.toml

```diff
@@ -96,8 +96,8 @@ rayon = "1.5"
 
 # Protobuf and gRPC
 prost = "0.12"
-tonic = { version = "0.10", default-features = false }
-tonic-build = { version = "0.10", default-features = false }
+tonic = { version = "0.11", default-features = false }
+tonic-build = { version = "0.11", default-features = false }
```

Comment:
[`tonic 0.11` bumps MSRV to 1.70](https://github.com/hyperium/tonic/blob/master/CHANGELOG.md#0110-2024-02-08). That also happens to be the next MSRV I wanted to target for `zcash_primitives` and its downstream crates (I'm more cautious about bumping MSRV for crates like `pasta_curves`, but for the crates in this workspace 1.70 should be fine). We were blocked from doing so by `zcashd` being stuck on 1.69, but that was finally fixed last month in https://github.com/zcash/zcash/pull/6879.

I note that in the first commit you avoided certain other dependency updates that would require 1.70; we can upgrade those now. Given that `tonic 0.11` only forces the case for `zcash_client_backend` with the relevant feature flag enabled, I'm fine with merging this PR as-is, and then landing a follow-up PR that uniformly bumps MSRV to 1.70.